source/Calamari.AzureWebApp/AzureWebAppBehaviour.cs

```diff
@@ -35,22 +36,30 @@ public bool IsEnabled(RunningDeployment context)
         public async Task Execute(RunningDeployment deployment)
         {
             var variables = deployment.Variables;
+            var hasJwt = !variables.Get(AzureAccountVariables.Jwt).IsNullOrEmpty();
             var subscriptionId = variables.Get(SpecialVariables.Action.Azure.SubscriptionId);
             var resourceGroupName = variables.Get(SpecialVariables.Action.Azure.ResourceGroupName, string.Empty);
             var siteAndSlotName = variables.Get(SpecialVariables.Action.Azure.WebAppName);
             var slotName = variables.Get(SpecialVariables.Action.Azure.WebAppSlot);
-
             var targetSite = AzureWebAppHelper.GetAzureTargetSite(siteAndSlotName, slotName);
-
-            var resourceGroupText = string.IsNullOrEmpty(resourceGroupName)
-                ? string.Empty
-                : $" in Resource Group '{resourceGroupName}'";
-            var slotText = targetSite.HasSlot
-                ? $", deployment slot '{targetSite.Slot}'"
-                : string.Empty;
+            var resourceGroupText = string.IsNullOrEmpty(resourceGroupName) ? string.Empty : $" in Resource Group '{resourceGroupName}'";
+            var slotText = targetSite.HasSlot ? $", deployment slot '{targetSite.Slot}'" : string.Empty;
+            var azureAccount = hasJwt ? (IAzureAccount)new AzureOidcAccount(variables) : new AzureServicePrincipalAccount(variables);
+            var accessToken = await azureAccount.GetAccessTokenAsync();
+            var isCurrentScmBasicAuthPublishingEnable = await AzureWebAppHelper.GetBasicPublishingCredentialsPoliciesAsync(subscriptionId, resourceGroupName, siteAndSlotName, accessToken.Token);
+            
+            if (!isCurrentScmBasicAuthPublishingEnable)
+            {
+                log.Info($"Detect 'SCM Basic Auth Publishing Credentials' configuration is setting off on '{targetSite.Site}'{slotText}. Turn it on for deployment and will set it off after finish deployment.");
```

Comment:
Is there a requirement for this? It feels like we should be leaving the SCM setting as is, and deploying depending on the setting, not updating it.